include/deal.II/grid/grid_tools.h

```diff
@@ -556,6 +556,34 @@ namespace GridTools
   get_active_neighbors (const typename Container::active_cell_iterator        &cell,
                         std::vector<typename Container::active_cell_iterator> &active_neighbors);
 
+  /**
+   * Extract the active cell layer around a subdomain (set of active cells) in
+   * the container @p container (i.e. those that share a common set of vertices)
+   * and return them in the vector @p active_halo_layer.
+   * This subdomain is defined according to the predicate @p predicate. This is
+   * expected to be a function that takes in an active cell an returns a boolean
+   * defining whether that cell belongs to the subdomain around which the halo
+   * layer is to be extracted.
+   *
+   * @tparam Container A type that satisfies the requirements of a mesh
+   * container (see
+   * @ref GlossMeshAsAContainer).
+   * @tparam Predicate A function that takes in an active cell an returns a boolean
+   * @param[in] container A mesh container (i.e. objects of type Triangulation,
+   * DoFHandler, or hp::DoFHandler)
+   * @param[in] predicate A function defining the subdomain around which the
+   * halo layer is to be extracted.
+   * @param[out] active_halo_layer A list of active cells sharing a common vertex
+   * with the predicated subdomain
+   *
+   * @author Jean-Paul Pelteret, Denis Davydov, 2015
+   */
+  template <class Container, typename Predicate>
+  void
+  compute_active_cell_halo_layer (const Container                                       &container,
+                                  const Predicate                                       &predicate,
+                                  std::vector<typename Container::active_cell_iterator> &active_halo_layer);
```

Comment:
A quick question regarding the return type. I see that `GridTools::find_active_cell_around_point` has a different return type definition for Visual Studio when compared to other compilers. Should I take this into account too?